package.json

```diff
@@ -4,6 +4,7 @@
 	"private": true,
 	"scripts": {
 		"worker": "npx wrangler dev --env local --test-scheduled",
+		"emulate": "vite build -w & npx wrangler pages dev .svelte-kit/cloudflare --live-reload --local --binding PRODUCTION=0 --d1 DB=DB",
```

Comment:
I'm guessing this `PRODUCTION=0` binds the local D1 database instead of the remote one?

Reply:
Okay 😭 so this is my cursed way of making cache invalidation compatible with the local environment so I could do a conditional statement for either `michigan-daily-alt-text-tracker.pages.dev` or `localhost:8788`, the local binding part for the db is --d1 DB=DB I think? I'm not sure why its different for `wrangler pages dev` but instead of being able to do `--env local` you have to do `--binding <env_var_name>=<env_var_value>`. 

Had to do this since I couldn't use SvelteKit's dev since it's running via wrangler. Also the binding variable is read as a string, so I used a 0 instead of false and parsed the int. I spent like 30 minutes before that trying to figure out why I was being gaslit into thinking "false" was true. Maybe on some level something like PRODUCTION === "false" would be more explicit though.

Reply:
Ok, that makes sense. I guess I'm curious how/where/if `PRODUCTION` is set to 1 in order for the correct URL to be selected when we're in production.

Reply:
I see that the worker has it set:

<img width="803" alt="Screenshot 2024-04-19 at 1 58 29 PM" src="https://github.com/michigandaily/alt-text-tracker/assets/44459660/ded6d4b2-5517-4fdc-9d64-93aeee4a6e6d">

But I don't see any variables for the website:

<img width="803" alt="Screenshot 2024-04-19 at 1 59 18 PM" src="https://github.com/michigandaily/alt-text-tracker/assets/44459660/7cc48813-7c09-4a9c-ba53-ff4fd6e79079">

Reply:
I think `PRODUCTION=true` in that case since it's specified in the `wrangler.toml `although it still might be read in as a string?? Which would evaluate to true anyways... I'm not sure. Although that does make the type mismatch kind of uncomfortable and weird. I'm thinking of just doing a string comparison instead

Reply:
maybe it would be better to just have a new "local" binding instead of trying to modify the PRODUCTION variable

Reply:
Oh hm just saw the variables for the website screenshot. True I guess env variables are different among worker and pages. I think I'll just set a local variable then and check if it exists

Reply:
Oh I see. I forgot about the `wrangler.toml` because it wasn't in this PR's changeset. I don't have strong feelings about this.

Reply:
Could you try setting the `PRODUCTION` variable in the dashboard for both preview and production?

Reply:
I don't think Pages supported `wrangler.toml` until recently: <https://blog.cloudflare.com/pages-workers-integrations-monorepos-nextjs-wrangler>. Since our toml file is older, I'm not sure its completely synced. I also don't know if we need to though.

Reply:
added PRODUCTION to preview and prod 👍

Reply:
Yeah I think we need to redeploy. I think there's a button somewhere to redeploy on the dashboard if you don't want to push.